src/lightning_app/utilities/imports.py

```diff
@@ -17,6 +17,23 @@
 from typing import List, Union
 
 from lightning_utilities.core.imports import module_available
+from packaging.requirements import Marker, Requirement
+
+try:
+    from importlib import metadata
+except ImportError:
+    # Python < 3.8
+    import importlib_metadata as metadata  # type: ignore
+
+
+def _get_extras(extras: str = "cloud") -> List[str]:
+    """Get the list of installable packages in the given extras."""
+    from lightning_app import __package_name__
+
+    extras = f"app-{extras}" if __package_name__ == "lightning" else extras
+    requirements = {r: Requirement(r) for r in metadata.requires(__package_name__)}
```

Comment:
Do we need a dict here instead of a list if we use r.name below?

Reply:
Yeah, `r.name` only has the package name and not any version limits